radixdlt-java/src/main/java/com/radixdlt/client/core/crypto/ECKeyPairGenerator.java

```diff
@@ -92,4 +93,43 @@ public ECKeyPair generateKeyPair(int numBits) {
 		}
 	}
 
+    /**
+     * Generates a new, deterministic {@code ECKeyPair} instance based on the
+     * provided seed.
+     *
+     * @param seed The seed to use when deriving the key pair instance.
+     * @return A key pair that corresponds to the provided seed.
+     * @throws IllegalArgumentException if the seed is empty or a null pointer.
+     */
+    public ECKeyPair generateKeyPairFromSeed(byte[] seed) {
```

Comment:
It feels a bit awkward to put this method in this class as it won't source an `ECKeyPair` from an elliptic curve, but rather from a `sha256` hash. The question can be taken further by asking if an `ECKeyPair` sourced from a hash really is an `ECKeyPair` or rather something else ("`HashKeyPair`"?)